target/linux/ath79/image/gen_kt_img.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+SUFFIX=.patched
+TMP_FILE1="/tmp/__tmp_file1__"
+TMP_FILE2="/tmp/__tmp_file2__"
+TMP_FILE3="/tmp/__tmp_file3__"
+
+function usage() {
+  echo ""
+  echo "$0 (bin file)"
+  echo ""
+}
+
+function patch_uimage_header() {
+  if [ ! -f ${1} ]
+  then
+    echo "File not found !"
+    exit 1
+  fi
+
+  dd if=${1} of=${TMP_FILE1} count=64 bs=1
+
+  dd if=${TMP_FILE1} of=${TMP_FILE2} count=4 bs=1
+  printf '\x00\x00\x00\x00\x00\x02\x04\x00' | dd of=${TMP_FILE2} bs=1 seek=4 conv=notrunc
+  dd bs=1 skip=8 if=${TMP_FILE1} >> ${TMP_FILE2}
+
+  CRC32_CHECKSUM=$(crc32 ${TMP_FILE2})
```

Comment:
You should replace crc32 to somewhat generic utility (or include the util itself in the patch) as we cannot be sure crc32 command will exist in other platforms.
This might be a good reference: https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/image/mt7621.mk#L51

Reply:
Thank you! I was able to add a header modifying funciton inside the Makefile thanks to the reference you provided!